xtask/src/main.rs

```diff
@@ -1,43 +1,50 @@
-use std::env::{self, Args};
+use clap::Parser;
+use run::run_with_sudo;
 
-use anyhow::Result;
-use xshell::{cmd, Shell};
+mod run;
 
-fn main() {
-    if let Err(e) = try_main() {
-        eprintln!("{}", e);
-        std::process::exit(-1);
-    }
+#[derive(Debug, Parser)]
+#[clap(disable_help_subcommand = true)]
+pub struct Options {
+    #[clap(subcommand)]
+    command: Command,
 }
 
-fn try_main() -> Result<()> {
-    let mut args = env::args();
-    let task = args.nth(1);
-    match task.as_deref() {
-        Some("test") => run_with_sudo("test-suite", &[], args)?,
-        Some("pulsard") => run_with_sudo("pulsar-exec", &["pulsard"], args)?,
-        Some("pulsar") => run_with_sudo("pulsar-exec", &["pulsar"], args)?,
-        Some("probe") => run_with_sudo("probe", &[], args)?,
-        _ => print_help(),
-    }
-    Ok(())
+#[derive(Debug, Parser)]
+enum Command {
+    /// Run a binary with admin privileges
+    Run {
+        #[clap(subcommand)]
+        command: Run,
+    },
```

Comment:
Not a deal-breaker, but I'm not a big fan the extra run subcommand:
- It's hard to differentiate what is "running" something and what not. Also the test-suite or eventual scripts are kind of run. I don't feel it gives any extra semantic meaning.
- It's more verbose
- If you don't have it, just running `cargo xtask` or `cargo xtask --help` will list all available commands. With run in the middle, you'll have to also run `cargo xtask run --help`

It's not a big deal though, we can keep it if you prefer.

Reply:
removed run keyword